src/impl/tokenParser.js

```diff
@@ -308,6 +314,10 @@ function dateTimeFromMatches(matches) {
     zone = null;
   }
 
+  if (!isUndefined(matches.q)) {
+    matches.M = (matches.q - 1) * 3 + 1;
```

Comment:
The unfortunate part of this is that the error message for specifying a quarter that isn't between 1-4 is that it will complain that the _months_ are out of range. I think we can live with that though. The alternative would be to make quarters a first-class unit in Luxon, which seems painful.

Reply:
I actually tried this:
```js
  if (!isUndefined(matches.q)) {
    if (matches.q < 1 || matches.q > 4)
    {
        throw new Error(`quarter value "${matches.q}" is out of 1-4 range`);
    }
    matches.M = (matches.q - 1) * 3 + 1;
  }
```
but thrown errors are not caught.

Reply:
Yeah, it's not the place where the validation happens; it happens later once the object has been assembled. That's the weird thing about Q just translating to M; the Q context gets lost by the time the validation step runs. But it is what is.